File: aplicacao/backend/Models/Result/TestCaseResult.cs
using Hl7.Fhir.Model;

namespace FHIRUT.API.Models.Result
{
    public class TestCaseResult
    {
        public string TestId { get; set; } = "";
        public string JsonId { get; set; } = "";
        public string ExpectedStatus { get; set; } = "";
        public string ActualStatus { get; set; } = "";
        public List<IssueSummary> Issues { get; set; } = new();
        public TimeSpan ExecutionTime { get; set; }
        public bool Passed { get; set; }
        public List<string> ValidationMessages { get; set; } = new();

        public OperationOutcome? OperationOutcome { get; set; }
    }
}

File: aplicacao/frontend/src/app/models/test-case-result.ts
import { OperationOutcome } from "fhir/r4";

export interface TestCaseResult {
  testId: string;
  jsonId: string;
  expectedStatus: string;
  actualStatus: string;
  issues: IssueSummary[];
  executionTime: string;
  operationOutcome?: OperationOutcome;
}

export interface IssueSummary {
  severity: string;
  location: string;
  code: string;
  details: string;
  source: string;
}
